root/edit/details/EditWork.js

```diff
@@ -0,0 +1,186 @@
+        +[attributeName: string]: $ReadOnlyArray<WorkAttributeT>,
+      },
+      +languages: $ReadOnlyArray<string>,
+    },
+  },
+  +display_data: {
+    +attributes?: {
+      +[attributeName: string]: CompT<$ReadOnlyArray<string>>,
+    },
+    +comment?: CompT<string | null>,
+    +iswc?: CompT<string | null>,
+    +language?: CompT<LanguageT | null>,
+    +languages?: CompT<$ReadOnlyArray<string>>,
+    +name?: CompT<string>,
+    +type?: CompT<WorkTypeT | null>,
+    +work: WorkT,
+  },
+};
+
+const EditWork = ({edit}: {edit: Props}) => {
+  const display = edit.display_data;
+  const comment = display.comment;
+  const iswc = display.iswc;
+  const name = display.name;
+  const type = display.type;
+  const language = display.language;
+  const languages = display.languages;
+  const attributes = display.attributes ?? {};
+  return (
+    <table className="details edit-work">
+      <tr>
+        <th>{addColonText(l('Work'))}</th>
+        <td colSpan="2">
+          <DescriptiveLink entity={display.work} />
+        </td>
+      </tr>
+      {name ? (
+        <WordDiff
+          label={addColonText(l('Name'))}
+          newText={name.new}
+          oldText={name.old}
+        />
+      ) : null}
+      {comment ? (
+        <WordDiff
+          label={addColonText(l('Disambiguation'))}
+          newText={comment.new ?? ''}
+          oldText={comment.old ?? ''}
+        />
+      ) : null}
+      {iswc ? (
+        <Diff
+          label={addColonText(l('ISWC'))}
+          newText={iswc.new ?? ''}
+          oldText={iswc.old ?? ''}
+        />
+      ) : null}
+      {type ? (
+        <FullChangeDiff
+          label={addColonText(l('Work type'))}
+          newContent={
+            type.new?.name ? lp_attributes(type.new.name, 'work_type') : ''
+          }
+          oldContent={
+            type.old?.name ? lp_attributes(type.old.name, 'work_type') : ''
+          }
+        />
+      ) : null}
+      {language ? (
+        <FullChangeDiff
+          label={addColonText(l('Language'))}
+          newContent={l_languages(language.new?.name ?? '')}
+          oldContent={l_languages(language.old?.name ?? '')}
+        />
+      ) : null}
+      {languages ? (
+        <Diff
+          label={addColonText(l('Lyrics Languages'))}
+          newText={commaOnlyListText(languages.new.map(
+            language => l_languages(language),
+          ))}
+          oldText={commaOnlyListText(languages.old.map(
+            language => l_languages(language),
+          ))}
+          split=", "
+        />
+      ) : null}
+      {edit.data.new.attributes
+        ? Object.keys(attributes).sort().map((attributeName) => (
+          <tr key={attributeName}>
+            <th>
+              {addColonText(lp_attributes(
+                attributeName,
+                'work_attribute_type',
+              ))}
+            </th>
+            <td className="old">
+              {attributes[attributeName].old.length > 0
```

Comment:
Since we access `attributes[attributeName]` at least six times, it'd be nice to cache the value as `attributeDiff` or something.